commons-mongo/src/main/scala/com/avsystem/commons/mongo/typed/MongoIndex.scala

```diff
@@ -0,0 +1,50 @@
+package com.avsystem.commons
+package mongo.typed
+
+import com.avsystem.commons.misc.{AbstractValueEnum, AbstractValueEnumCompanion, EnumCtx}
+import org.bson.{BsonDocument, BsonValue}
+
+
+case class MongoIndex[E](fields: Vector[(MongoPropertyRef[E, _], MongoIndexType)]) {
```

Comment:
What about index properties? https://docs.mongodb.com/manual/core/index-properties/

Reply:
These are passed separately in `IndexOptions`.

Yes, we could rewrite `MongoIndex` to better express various index types along with their options but I decided it would be too much work for now.